src/main/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -580,94 +613,106 @@ public void finish() {
             metrics.PCT_EXC_OVERLAP        /= (double) metrics.PF_BASES_ALIGNED;
             metrics.PCT_EXC_OFF_TARGET     /= (double) metrics.PF_BASES_ALIGNED;
 
-            // Get Theoretical Het SNP Sensitivity and Calculate Target Coverage Metrics
-            final double[] coverageDistribution = calculateTargetCoverageMetrics();
-
-            // Construct and write the outputs
-            for (int i=0; i<baseQHistogramArray.length; ++i) {
-                baseQHistogram.increment(i, baseQHistogramArray[i]);
-            }
-
-            // Construct and write the outputs
-            for (int i = 0; i < coverageDistribution.length; ++i) {
-                depthHistogram.increment(i, coverageDistribution[i]);
-            }
-
-            final double [] depthDoubleArray = TheoreticalSensitivity.normalizeHistogram(depthHistogram);
-            final double [] baseQDoubleArray = TheoreticalSensitivity.normalizeHistogram(baseQHistogram);
-            metrics.HET_SNP_SENSITIVITY = TheoreticalSensitivity.hetSNPSensitivity(depthDoubleArray, baseQDoubleArray, sampleSize, LOG_ODDS_THRESHOLD);
-            metrics.HET_SNP_Q = QualityUtil.getPhredScoreFromErrorProbability((1 - metrics.HET_SNP_SENSITIVITY));
-
+            calculateTargetCoverageMetrics();
+            calculateTheoreticalHetSensitivity();
             calculateGcMetrics();
             emitPerBaseCoverageIfRequested();
         }
 
         /** Calculates how much additional sequencing is needed to raise 80% of bases to the mean for the lane. */
-        private double[] calculateTargetCoverageMetrics() {
-            final int[] depths = new int[(int) this.metrics.TARGET_TERRITORY];
```

Comment:
Just to reiterate, the reduction in memory comes from getting rid of `depths` array of size `TARGET_TERRITORY`. The increase in memory comes from adding unfiltered histograms for het sensitivity, and it's independent of this reduction in memory. In other words, if I didn't add the new histograms, we'd just get the memory reduction for free.